twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -113,9 +117,11 @@ public Integer apply(BitSet input) {
   private final Location applicationLocation;
   private final Set<String> runnableNames;
   private final Map<String, Map<String, String>> logLevels;
+  private final Map<String, Integer> maxRetries;
```

Comment:
from my analysis the association of instance ids does not necessarily correspond to specific processes.  it looked like there is a pool of requests, a new request being serviced gets the lowest instance id and a failed request gets put back on the queue.  this is why i went with an instance adjusted number of retries.  did i miss something?